File: LeetCodeProblemsGo/LeetCode386LexicographicalNumbers/LeetCode386LexicographicalNumbers.go
package leetcode386lexicographicalnumbers

func lexicalOrder(n int) []int {
	result := []int{}
	for i := 1; i < 10; i++ {
		dfs(&result, i, n)
	}
	return result
}
func dfs(result *[]int, current int, n int) {
	if current > n {
		return
	}
	*result = append(*result, current)
	current *= 10
	next := 0
	for i := 0; i < 10; i++ {
		next = current + i
		if current > n {
			return
		}
		dfs(result, next, n)
	}
}

File: LeetCodeProblemsCMake/Problems/LeetCode386LexicographicalNumbers.cpp
//
// Created by saeed on 6/8/25.
//

#include "LeetCode386LexicographicalNumbers.h"
using namespace std;
void Dfs( vector<int> & result,int n,int currentNum) {

    if (currentNum>n)
        return;
    result.push_back(currentNum);
    currentNum*=10;
    int next{};
    for (int i=0;i<=9;i++) {
        next=currentNum+i;
        if (next>n)
            return;
        Dfs(result,n,next);
    }
}
 vector<int> LeetCode386LexicographicalNumbers::lexicalOrder(int n) {
    vector<int> result{};
     for (int i=1;i<=9;i++) {
         Dfs(result,n,i);
     }
    return result;
}
